backend/ecs_tasks/delete_files/main.py

```diff
@@ -141,21 +144,26 @@ def execute(queue_url, message_body, receipt_handle):
         input_bucket, input_key = parse_s3_url(object_path)
         validate_bucket_versioning(client, input_bucket)
         match_ids = build_matches(cols, manifest_object)
-        s3 = s3fs.S3FileSystem(
-            session=session,
-            default_cache_type="none",
-            requester_pays=True,
-            default_fill_cache=False,
-            version_aware=True,
+        s3 = pa.fs.S3FileSystem(
+            region=os.getenv("AWS_DEFAULT_REGION"),
+            session_name=ROLE_SESSION_NAME,
+            external_id=ROLE_SESSION_NAME,
+            role_arn=body.get("RoleArn"),
+            load_frequency=60 * 60,
         )
         # Download the object in-memory and convert to PyArrow NativeFile
         logger.info("Downloading and opening %s object in-memory", object_path)
-        metadata = s3.metadata(object_path, refresh=True)
-        with s3.open(object_path, "rb") as f:
-            source_version = f.version_id
+        with s3.open_input_stream(
+            "{}/{}".format(input_bucket, input_key), buffer_size=FIVE_MB
```

Comment:
Buffer size is same as we used to do it with s3fs: https://github.com/fsspec/s3fs/blob/main/s3fs/core.py#L246